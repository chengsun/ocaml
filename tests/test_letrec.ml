let rec parity = function
  | 0 -> Printf.printf "even!"
  | n -> parity_odd (pred n)
and parity_odd = function
  | 0 -> Printf.printf "odd!"
  | n -> parity (pred n)


(*
let closure_parity n =
  let rec parity = function
    | 0 -> Printf.printf "%d is even!" n
    | n -> parity_odd (pred n)
  and parity_odd = function
    | 0 -> Printf.printf "%d is odd!" n
    | n -> parity (pred n)
  in
  parity n
*)

let () = (
  parity 4;
  parity 5;
  (*
  closure_parity 4;
  closure_parity 5;
  *)
  ()
)
