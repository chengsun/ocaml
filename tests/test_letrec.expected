even!
odd!
4 is even!
5 is odd!
